wit/stream-consumer.wit

```diff
@@ -0,0 +1,7 @@
+/// A stream consumer, commonly used for event streaming.
+default interface stream-consumer {
+	use pkg.types.{message, error, receiver}
+
+	/// `receive-message` satisfied pull-based messaging requirements
+	read-message(r: receiver, partition: u32, offset: u32) -> result<message, error>
+}
```

Comment:
the consumer needs to be set up with a configuration that specifies things like partitions and offsets and subsequently  just read. 

part of that setup needs to be a configuration of a) a checkpoint abstraction and b) a lease abstraction for whether the client still owns the partition. 

I expect future event streams to kill partitions altogether as a user-facing concept. You should also anticipate "offset" to become a query. "u32" is already false for EH and Kafka, which allow for start,end,time.

This needs work.